stdlib/public/Concurrency/Actor.cpp

```diff
@@ -832,28 +832,38 @@ class alignas(sizeof(void *) * 2) ActiveActorStatus {
 
 /// Given that a job is enqueued normally on a default actor, get/set
 /// the next job in the actor's queue.
-static JobRef getNextJobInQueue(Job *job) {
+static JobRef getNextJobRef(Job *job) {
   return *reinterpret_cast<JobRef *>(job->SchedulerPrivate);
 }
-static void setNextJobInQueue(Job *job, JobRef next) {
+static void setNextJobRef(Job *job, JobRef next) {
   *reinterpret_cast<JobRef *>(job->SchedulerPrivate) = next;
 }
 
-namespace {
+static Job* getNextJob(Job *job) {
+  return getNextJobRef(job).getAsPreprocessedJob();
+}
+static void setNextJob(Job *job, Job* next) {
+  setNextJobRef(job, JobRef::getPreprocessed(next));
+}
 
-struct JobQueueTraits {
-  static Job *getNext(Job *job) {
-    return getNextJobInQueue(job).getAsPreprocessedJob();
-  }
-  static void setNext(Job *job, Job *next) {
-    setNextJobInQueue(job, JobRef::getPreprocessed(next));
-  }
-  static int compare(Job *lhs, Job *rhs) {
-    return descendingPriorityOrder(lhs->getPriority(), rhs->getPriority());
+enum { NumPriorityBuckets = 5 };
+
+static int getPriorityIndex(JobPriority priority) {
+  switch (priority) {
+    case JobPriority::UserInteractive:
+      return 0;
+    case JobPriority::UserInitiated:
+      return 1;
+    case JobPriority::Unspecified:
+      assert(false);
+    case JobPriority::Default:
+      return 2;
+    case JobPriority::Utility:
+      return 3;
+    case JobPriority::Background:
+      return 4;
```

Comment:
I'm comfortable with only using 5 buckets and not trying to properly order the other values within buckets, but let's make sure we at least reliably map them all to a bucket.  The compiler will probably optimize a series of comparisons fairly well; you could also just precompute it into a lookup table with some `constexpr` magic.

Reply:
Just a lookup table won't work, because unknown priorities, being unknown, have no upper bound. Updated to `if-else-if` chain.

Reply:
Well, there's only 8 bits of priority in the Job header.